src/api/procedures/modifyInstructionAuthorization.ts

```diff
@@ -0,0 +1,98 @@
+import BigNumber from 'bignumber.js';
+
+import { Instruction } from '~/api/entities';
+import { checkInstructionValidation } from '~/api/procedures/utils';
+import { PolymeshError, Procedure } from '~/base';
+import { AuthorizationStatus, ErrorCode, InstructionAuthorizationOperation } from '~/types';
+import {
+  meshAuthorizationStatusToAuthorizationStatus,
+  numberToU64,
+  portfolioIdToMeshPortfolioId,
+} from '~/utils';
+
+export interface ModifyInstructionAuthorizationParams {
+  operation: InstructionAuthorizationOperation;
+}
+
+/**
+ * @hidden
+ */
+export type Params = ModifyInstructionAuthorizationParams & {
+  id: BigNumber;
+};
+
+/**
+ * @hidden
+ */
+export async function prepareModifyInstructionAuthorization(
```

Comment:
Why are you only returning the instruction in 2/3 cases?

Reply:
I thought if operation is to reject the Instruction, return it seems some weird

Reply:
Makes sense, but rejecting is not deleting. The instruction can still be re-authorized in some cases and the pertinent data still exists